spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,549 @@
+`Float`, `String`, `Boolean`, `Null`, or `Enum`), a list (`ListValue`), an input object
+(`ObjectValue`), or a `Variable`.
+
+Within the scope of the {FieldSelectionMap}, the relationship between input and output is
+established by defining the `Value` of the argument as a selection of fields from the output object.
+
+Yet only certain types of `Value` have a semantic meaning. 
+`ObjectValue` and `ListValue` are used to define the structure of the value. 
+Scalar values, on the other hand, do not carry semantic importance in this context, and variables
+are excluded as they do not exist. 
+Given that these potential values do not align with the standard literals defined in the GraphQL
+specification, a new literal called {SelectedValue} is introduced, along with {SelectedObjectValue},
+
+Beyond these literals, an additional literal called {Path} is necessary.
+
+### Name
+
+Is equivalent to the {Name} defined in the 
+[GraphQL specification](https://spec.graphql.org/October2021/#Name)
+
+### Path
+Path :: 
+    - FieldName
+    - Path . FieldName
+    - FieldName < TypeName > . Path
+
+FieldName ::
+    - Name
+
+TypeName ::
+    - Name
+
+The {Path} literal is a string used to select a single output value from the _return type_ by
+specifying a path to that value. 
+This path is defined as a sequence of field names, each separated by a period (`.`) to create 
+segments. 
+
+``` example
+book.title
+```
+
+Each segment specifies a field in the context of the parent, with the root segment referencing a
+field in the _return type_ of the query. 
+Arguments are not allowed in a {Path}.
+
+To select a field when dealing with abstract types, the segment selecting the parent field must 
+specify the concrete type of the field using angle brackets after the field name if the field is not 
+defined on an interface.
+
+In the following example, the path `mediaById.<Book>.isbn` specifies that `mediaById` returns a
+`Book`, and the `isbn` field is selected from that `Book`.
+
+``` example
+mediaById<Book>.isbn
+```
+
+### SelectedValue
+SelectedValue :: 
+    - Path
+    - SelectedObjectValue
+    - Path . SelectedObjectValue
+    - SelectedValue | SelectedValue 
+
+A {SelectedValue} is defined as either a {Path} or a {SelectedObjectValue}
+
+A {Path} is designed to point to only a single value, although it may reference multiple fields
+depending on the return type. To allow selection from different paths based on
+type, a {Path} can include multiple paths separated by a pipe (`|`).
+
+In the following example, the value could be `title` when referring to a `Book` and `movieTitle`
+when referring to a `Movie`. 
+
+``` example
+mediaById<Book>.title | mediaById<Movie>.movieTitle
+```
+
+The `|` operator can be used to match multiple possible {SelectedValue}. This operator is applied
+when mapping an abstract output type to a `@oneOf` input type.
+
+```example
+{ movieId: <Movie>.id } | { productId: <Product>.id }
```

Comment:
Is `<Movie>.id` valid on it's own? Don't we need a `FieldName` first?

```
{ movieId: media<Movie>.id } | { productId: media<Product>.id }
```

Reply:
maybe we do requrie the `PathSegment` after all:
```
Path :: 
    - < TypeName > . PathSemgent
    - PathSemgent

PathSegment :: 
    - FieldName
    - FieldName . PathSemgent
    - FieldName < TypeName > . PathSemgent

FieldName ::
    - Name

TypeName ::
    - Name
    ```

Reply:
What about ?

```
Path :: 
    - FieldName
    - FieldName < TypeName >
    - Path . Path 

FieldName ::
    - Name

TypeName ::
    - Name
```

Reply:
i will go with the suggestin above for now